CopyQ.py

```diff
@@ -55,21 +55,48 @@
 
 def handleQuery(query):
     if query.isTriggered:
+        try:
+            pattern = re.compile(query.string, re.IGNORECASE)
+        except re.error:
+            return [
+                Item(
+                    id=__prettyname__,
+                    icon=iconPath,
+                    text="<b>Invalid regex!</b>",
+                    subtext="",
+                    actions=[],
+                )
+            ]
 
-        script = copyq_script_getMatches % query.string if query.string else copyq_script_getAll
+        escaped_query = query.string.replace('\\', '\\\\').replace('"', '\\"')
+        script = copyq_script_getMatches % escaped_query if query.string else copyq_script_getAll
 
         proc = subprocess.run(['copyq', '-'], input=script.encode(), stdout=subprocess.PIPE)
         json_arr = json.loads(proc.stdout.decode())
 
+        if not json_arr:
+            return [
+                Item(
+                    id=__prettyname__,
+                    icon=iconPath,
+                    text="<b>No results found.</b>",
+                    subtext="",
+                    actions=[],
+                )
+            ]
+
         items = []
-        pattern = re.compile(query.string, re.IGNORECASE)
         for json_obj in json_arr:
             row = json_obj['row']
             text = json_obj['text']
             if not text:
                 text = "<i>No text</i>"
             else:
-                text = pattern.sub(lambda m: "<u>%s</u>" % m.group(0), html.escape(" ".join(filter(None, text.replace("\n", " ").split(" ")))))
+                U_BEGIN = "@U_BEGIN_XXXXX@"
+                U_END = "@U_END_XXXXX@"
+                raw_text = " ".join(filter(None, text.replace("\n", " ").split(" ")))
+                underlined_text = pattern.sub(lambda m: U_BEGIN + m.group(0) + U_END, raw_text)
+                text = html.escape(underlined_text).replace(U_BEGIN, "<u>").replace(U_END, "</u>")
```

Comment:
why this indirection over U_BEGIN and U_END?

Reply:
The problem is that if you have copied `<u></u>` (and other tags), we want to escape them so we don't view them raw.
We still want to add `<u></u>` to the parts of the string that matched the query, so we need this indirection.